ingestion/src/metadata/ingestion/source/database/saperp/metadata.py

```diff
@@ -0,0 +1,297 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+SAP ERP source module
+"""
+import traceback
+from typing import Iterable, List, Optional
+
+from metadata.generated.schema.api.data.createTable import CreateTableRequest
+from metadata.generated.schema.entity.data.databaseSchema import DatabaseSchema
+from metadata.generated.schema.entity.data.table import (
+    Column,
+    ColumnName,
+    Constraint,
+    ConstraintType,
+    DataType,
+    Table,
+    TableConstraint,
+    TableType,
+)
+from metadata.generated.schema.entity.services.connections.database.sapErpConnection import (
+    SapErpConnection,
+)
+from metadata.generated.schema.entity.services.ingestionPipelines.status import (
+    StackTraceError,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.basic import (
+    EntityName,
+    FullyQualifiedEntityName,
+    Markdown,
+)
+from metadata.ingestion.api.models import Either
+from metadata.ingestion.api.steps import InvalidSourceException
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.source.database.column_type_parser import ColumnTypeParser
+from metadata.ingestion.source.database.common_db_source import CommonDbSourceService
+from metadata.ingestion.source.database.saperp.constants import TABLE_TYPE_MAP
+from metadata.ingestion.source.database.saperp.models import (
+    ColumnsAndConstraints,
+    SapErpColumn,
+    SapErpTable,
+    TableConstraintsModel,
+)
+from metadata.utils import fqn
+from metadata.utils.execution_time_tracker import calculate_execution_time_generator
+from metadata.utils.filters import filter_by_table
+from metadata.utils.helpers import clean_up_starting_ending_double_quotes_in_string
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class SaperpSource(CommonDbSourceService):
```

Comment:
Is there a specific reason we are not using the `CommonNoSQLSource` instead of `CommonDbSourceService`. It seems more suitable for this kind source that does not support SQL.

Reply:
Makes sense I think we can convert it to NoSQLSource

Reply:
@sushi30 upon checking this, the connector would makes sense to use the SQLSource since SAP ERP tables can use SQL for querying.
We use API to get the metadata out but the system is based on SQL itself

Reply:
I dont see a reference to SQLalchemy anywhere in the submitted code. In this case we might be duplicating a lot of logic from the NoSQL source which generally uses APIs to extract metadata (things like filters). In any case I dont think its a blocker so I approved.